Use cfg=kani_host for host crates (#3244)
model-checking/kani: tests/cargo-kani/build-rs-plus-host-with-kani-proofs/README.md

> This repo contains contains a minimal example that breaks compilation when using [kani](https://github.com/model-checking/kani), where I would expect compilation to work.

Reviewer: ```suggestion
This repository contains a minimal example that breaks compilation when using [kani](https://github.com/model-checking/kani), where we would expect compilation to work.
```

Deleting the `binary/build.rs` script makes the compilation work suddenly, despite it being skipped anyways:

```
binary$ cargo kani -v
Kani Rust Verifier 0.48.0 (cargo plugin)

> Skipped the following unsupported targets: 'build-script-build'.

Author: Admitted I also don't know what "build-script-build" means - could you enlighten me so that I can add specific text for this case (rather than throwing it into the catch-all `ignored-unsupported`?

> Skipped the following unsupported targets: 'build-script-build'.

Reviewer 2: That target represents the `build.rs` file. https://doc.rust-lang.org/cargo/reference/build-scripts.html
